lib/api/include/irods/generalUpdate.h

```diff
@@ -1,12 +1,18 @@
-#ifndef GENERAL_UPDATE_H__
-#define GENERAL_UPDATE_H__
+#ifndef IRODS_GENERAL_UPDATE_H
+#define IRODS_GENERAL_UPDATE_H
 
 #include "irods/rcConnect.h"
 #include "irods/rodsGeneralUpdate.h"
 
 #ifdef __cplusplus
-extern "C"
+extern "C" {
 #endif
-int rcGeneralUpdate( rcComm_t *conn, generalUpdateInp_t *generalUpdateInp );
 
+__attribute__((deprecated("Its use should be avoided."))) int rcGeneralUpdate(rcComm_t* conn,
```

Comment:
perhaps just `deprecated("Avoid use of this function.")` ?

Reply:
Here's an alternative: `GeneralUpdate is deprecated. Its use should be avoided.`

That could work since `GeneralUpdate` can be viewed as a group of related things.

Thoughts?

Reply:
That sounds good. This is in line with how `SimpleQuery` was deprecated: https://github.com/irods/irods/commit/185302df12b2883497bcf7d0928fbc39e04a4668

Reply:
Done.